pkg/netlink/netlinkstate.go

```diff
@@ -0,0 +1,791 @@
+			neighborRoutes = append(neighborRoutes, R0)
+		}
+	}
+	return neighborRoutes
+}
+
+// ParseRoute parse the routes
+// nolint
+func ParseRoute(v *infradb.Vrf, Rm []routeCmdInfo, t int) routeList {
+	var route routeList
+	for _, Ro := range Rm {
+		if Ro.Type == "" && (Ro.Dev != "" || Ro.Gateway != "") {
+			Ro.Type = routeTypeLocal
+		}
+		var rs RouteStruct
+		rs.Vrf = v
+		if Ro.Nhid != 0 || Ro.Gateway != "" || Ro.Dev != "" {
+			rs.Nexthops = append(rs.Nexthops, NHParse(v, Ro))
+		}
+		rs.NlType = "unknown"
+		rs.Route0.Table = t
+		rs.Route0.Priority = 1
+		if Ro.Dev != "" {
+			dev, _ := vn.LinkByName(Ro.Dev)
+			rs.Route0.LinkIndex = dev.Attrs().Index
+		}
+		if Ro.Dst != "" {
+			var Mask int
+			split := Ro.Dst
+			if strings.Contains(Ro.Dst, "/") {
+				split4 := strings.Split(Ro.Dst, "/")
+				Mask, _ = strconv.Atoi(split4[1])
+				split = split4[0]
+			} else {
+				Mask = 32
+			}
+			var nIP *net.IPNet
+			if Ro.Dst == "default" {
+				nIP = &net.IPNet{
+					IP:   net.ParseIP("0.0.0.0"),
+					Mask: net.IPv4Mask(0, 0, 0, 0),
+				}
+			} else {
+				mtoip := netMaskToInt(Mask)
+				b3 := make([]byte, 8) // Converting int64 to byte
+				binary.LittleEndian.PutUint64(b3, uint64(mtoip[3]))
+				b2 := make([]byte, 8)
+				binary.LittleEndian.PutUint64(b2, uint64(mtoip[2]))
+				b1 := make([]byte, 8)
+				binary.LittleEndian.PutUint64(b1, uint64(mtoip[1]))
+				b0 := make([]byte, 8)
+				binary.LittleEndian.PutUint64(b0, uint64(mtoip[0]))
+				nIP = &net.IPNet{
+					IP:   net.ParseIP(split),
+					Mask: net.IPv4Mask(b0[0], b1[0], b2[0], b3[0]),
+				}
+			}
+			rs.Route0.Dst = nIP
+		}
+		if Ro.Metric != 0 {
+			rs.Route0.Priority = Ro.Metric
+		}
+		if Ro.Protocol != "" {
+			if rtnProto[Ro.Protocol] != 0 {
+				rs.Route0.Protocol = vn.RouteProtocol(rtnProto[Ro.Protocol])
+			} else {
+				rs.Route0.Protocol = 0
+			}
+		}
+		if Ro.Type != "" {
+			rs.Route0.Type = rtnType[Ro.Type]
+		}
+		if len(Ro.Flags) != 0 {
+			rs.Route0.Flags = getFlag(Ro.Flags[0])
+		}
+		if Ro.Scope != "" {
+			rs.Route0.Scope = vn.Scope(rtnScope[Ro.Scope])
+		}
+		if Ro.Prefsrc != "" {
+			nIP := &net.IPNet{
+				IP: net.ParseIP(Ro.Prefsrc),
+			}
+			rs.Route0.Src = nIP.IP
+		}
+		if Ro.Gateway != "" {
+			nIP := &net.IPNet{
+				IP: net.ParseIP(Ro.Gateway),
+			}
+			rs.Route0.Gw = nIP.IP
+		}
+		if Ro.VRF != nil {
+			rs.Vrf, _ = infradb.GetVrf(Ro.VRF.Name)
+		}
+		if Ro.Table != 0 {
+			rs.Route0.Table = Ro.Table
+		}
+		rs.NlType = setRouteType(&rs, v)
+		rs.Key = routeKey{Table: rs.Route0.Table, Dst: rs.Route0.Dst.String()}
+		if preFilterRoute(&rs) {
+			route.RS = append(route.RS, &rs)
+		}
+	}
+	return route
+}
+
+// checkRoute checks the route
+func checkRoute(r *RouteStruct) bool {
+	Rk := r.Key
+	for k := range latestRoutes {
+		if k == Rk {
+			return true
+		}
+	}
+	return false
+}
+
+// deleteNH deletes the neighbor
+func deleteNH(nexthop []*NexthopStruct) []*NexthopStruct {
```

Comment:
Pls explain what was the intention here? If 1 element then return empty slice, if there are more, then return the last?
```go
l := len(nexthop)
if l > 1 {
  return []*Nexthop{nexthop[l - 1]}
}
return  []*Nexthop{}
```

Reply:
at this moment it looks like
```go
func deleteNH(nexthop []*NexthopStruct) []*NexthopStruct {
	return nexthop[:0]
}
```
1. It is not used within this PR. Is it needed at all?
2. If you want to return an empty slice, you can just use `[]*NexthopStruct{}`
3. If you want to clear the passed slice `nexthop = []*NexthopStruct{}`

Reply:
used []*NexthopStruct{}, removed deleteNH which is not used anymore